crates/sui-core/src/authority.rs

```diff
@@ -895,6 +896,78 @@ impl ExecutionEnv {
     }
 }
 
+#[derive(Debug)]
+pub struct ForkRecoveryState {
+    /// Transaction digest to effects digest overrides
+    transaction_overrides: RwLock<HashMap<TransactionDigest, TransactionEffectsDigest>>,
+    /// Checkpoint sequence to checkpoint digest overrides  
+    checkpoint_overrides: RwLock<HashMap<CheckpointSequenceNumber, CheckpointDigest>>,
```

Comment:
we should use `parking_lot::RwLock` here instead of `tokio::sync::RwLock`, since the locks do not need to be held across `await` points